river/ensemble/streaming_random_patches.py

```diff
@@ -568,3 +353,690 @@ def random_subspace(all_features: list, k: int, rng: np.random.Generator):
+        model: base.Regressor = None,
+        n_models: int = 100,
+        subspace_size: typing.Union[int, float, str] = 0.6,
+        training_method: str = "patches",
+        lam: float = 6.0,
+        drift_detector: base.DriftDetector = None,
+        warning_detector: base.DriftDetector = None,
+        disable_detector: str = "off",
+        disable_weighted_vote: bool = True,
+        drift_detection_criteria: str = "error",
+        aggregation_method: str = "mean",
+        seed=None,
+        metric: RegressionMetric = None,
+    ):
+
+        # Check arguments for parent class
+        if model is None:
+            model = HoeffdingTreeRegressor(grace_period=50, split_confidence=0.01)
+
+        if drift_detector is None:
+            drift_detector = ADWIN(delta=1e-5)
+
+        if warning_detector is None:
+            warning_detector = ADWIN(delta=1e-4)
+
+        if disable_detector == "off":
+            pass
+        elif disable_detector == "drift":
+            drift_detector = None
+            warning_detector = None
+        elif disable_detector == "warning":
+            warning_detector = None
+        else:
+            raise AttributeError(
+                f"{disable_detector} is not a valid value for disable_detector.\n"
+                f"Valid options are: 'off', 'drift', 'warning'"
+            )
+
+        if metric is None:
+            metric = MAE()
+
+        super().__init__(
+            model=model,
+            n_models=n_models,
+            subspace_size=subspace_size,
+            training_method=training_method,
+            lam=lam,
+            drift_detector=drift_detector,
+            warning_detector=warning_detector,
+            disable_detector=disable_detector,
+            disable_weighted_vote=disable_weighted_vote,
+            seed=seed,
+            metric=metric,
+        )
+
+        if aggregation_method not in {self._MEAN, self._MEDIAN}:
+            raise ValueError(
+                f"Invalid aggregation_method: {aggregation_method}.\n"
+                f"Valid options are: {[self._MEAN, self._MEDIAN]}"
+            )
+        self.aggregation_method = aggregation_method
+
+        if drift_detection_criteria not in {self._ERROR, self._PREDICTION}:
+            raise ValueError(
+                f"Invalid drift_detection_criteria: {drift_detection_criteria}.\n"
+                f"Valid options are: {[self._ERROR, self._PREDICTION]}"
+            )
+        self.drift_detection_criteria = drift_detection_criteria
+
+        self._base_learner_class = BaseSRPRegressor
+
+    @classmethod
+    def _unit_test_params(cls):
+        return {"n_models": 3}
+
+    def _unit_test_skips(self):
+        return {
+            "check_shuffle_features_no_impact",
+            "check_emerging_features",
+            "check_disappearing_features",
+        }
+
+    def learn_one(self, x: dict, y: base.typing.RegTarget, **kwargs):
+        return super().learn_one(x=x, y=y, **kwargs)
+
```

Comment:
Is `learn_one` implemented here to account for different signatures?

Reply:
Yes, to indicate the type of the target value.